### ISSUE TITLE
ccm CLI unable to display profile even if world_readable = 1

### ISSUE THREAD

**Issue by ned21** [quattor/CCM#104]
```
$ ccm --comp dirperm
Uncaught exception!!! Calling stack is:
  LC::Exception::throw_error called at /usr/lib/perl/EDG/WP4/CCM/Configuration.pm line 472
  EDG::WP4::CCM::Configuration::elementExists called at /usr/lib/perl/EDG/WP4/CCM/CLI.pm line 104
  EDG::WP4::CCM::CLI::action_show called at /usr/lib/perl/EDG/WP4/CCM/Options.pm line 375
  EDG::WP4::CCM::Options::action called at /usr/sbin/ccm line 49
*** EDG::WP4::CCM::Element->elementExists (EDG::WP4::CCM::Configuration=HASH(0x256fd68),
/software/components/dirperm): failed to open DB_File /var/lib/ccm/profile.547/path2eid.db for read: Permission denied
$ grep world_readable /etc/ccm.conf
world_readable 1
$ ls -l /var/lib/ccm/profile.547/
total 1472
-rw-r--r-- 1 root root      0 May 26 05:10 ccm-active-profile.547-26218
-rw-r----- 1 root root 675840 May 17 09:12 eid2data.db
-rw-r--r-- 1 root root      8 May 17 09:12 eid2data.fmt
-rw-r----- 1 root root 663552 May 17 09:12 path2eid.db
-rw-r--r-- 1 root root      8 May 17 09:12 path2eid.fmt
-rw-r--r-- 1 root root     64 May 17 09:12 profile.url
-rw-r--r-- 1 root root 387397 May 17 09:12 profile.xml
$ rpm -q ccm
ccm-16.2.0-1.noarch
```

Looks like whatever writes the .db files needs to ensure it uses the correct perms.

**Comment by stdweird** [comment]
any chance you can try to install lastest master from https://jenkins1.ugent.be/view/Quattor/job/CCM/org.quattor.client.libraries$ccm/lastSuccessfulBuild/artifact/org.quattor.client.libraries/ccm/16.2.1-SNAPSHOT/, do a `ccm-fetch --force` and try again (incl dthe `ls -l`)? (it's ok if it downgrades ccm again, but will trigger restarts of ncm-cdispd and friends, so you have been warned :smile: )
the was some permission work done for the group readable option, maybe it also fixed other things.
(but if it is a db file, i guess this some umask setting that is wrong, but also those have been changed).

**Comment by ned21** [comment]
It depends on HEAD of CAF too:

```
$ ccm-fetch --force
CAF::Kerberos does not define $CAF::Kerberos::VERSION--version check failed at /usr/lib/perl/EDG/WP4/CCM/Fetch/Download.pm line 44.
BEGIN failed--compilation aborted at /usr/lib/perl/EDG/WP4/CCM/Fetch/Download.pm line 44.
Compilation failed in require at /usr/share/perl5/parent.pm line 20.
BEGIN failed--compilation aborted at /usr/lib/perl/EDG/WP4/CCM/Fetch.pm line 61.
Compilation failed in require at /usr/sbin/ccm-fetch line 31.
BEGIN failed--compilation aborted at /usr/sbin/ccm-fetch line 31.
```

and Perl-CAF requires a new dependency:
```
# sudo rpm -U perl-CAF-16.2.1-SNAPSHOT.rpm
error: Failed dependencies:
        perl(enum) is needed by perl-CAF-16.2.1-SNAPSHOT20160519115057.noarch
```
although fortunately that seems to be available in our repos. :)

Problem persists:

```
# ls -l /var/lib/ccm/profile.548/
total 1728
-rw-r----- 1 root root 675840 May 27 04:53 eid2data.db
-rw-r--r-- 1 root root      8 May 27 04:53 eid2data.fmt
-rw-r----- 1 root root 663552 May 27 04:53 path2eid.db
-rw-r--r-- 1 root root      8 May 27 04:53 path2eid.fmt
-rw-r--r-- 1 root root     64 May 27 04:53 profile.url
-rw-r--r-- 1 root root 387397 May 27 04:53 profile.xml
-rw-r--r-- 1 root root 261229 May 27 04:53 tabcompletion
# ccm-fetch --version
This is ccm-fetch version 16.2.1-SNAPSHOT
```

**Comment by stdweird** [comment]
@ned21 out of curiosity, what was the format (i.e. the content of path2eid.fmt)? old code had 0640 for DB and GDBM, and whatever umask allowed for CDB. 
this is now fixed in #106  (code was to messy not to clean it up, and usage of CAF::Path requires a class anyway)
